src/components/KSkeleton/FullScreenGenericSpinner.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+  <div
+    class="fullscreen-loading-container"
+    data-testid="full-screen-loader"
+  >
+    <div>
+      <div
+        class="loader"
+      />
+      <div
+        v-if="!hideProgress"
+        class="progress"
+      >
+        <div
+          class="progress-bar"
+          role="progressbar"
+          :style="{ width: `${progression}%` }"
+          title="Loading"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, computed, onMounted, onUnmounted } from 'vue'
+
+export default defineComponent({
+  name: 'FullScreenGenericSpinner',
+  props: {
+    progress: {
+      type: Number,
+      default: null,
+    },
+    hideProgress: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  setup(props) {
+    const timer = ref(0)
+    const progressInternal = ref(0)
+    const progression = computed(() => props.progress !== null ? props.progress : progressInternal.value)
+
+    onMounted(() => {
+      // If progress is user determined, don't fire interval to simulate progress
+      if (props.progress) {
+        return
+      }
+      timer.value = setInterval(() => {
+        if (progressInternal.value >= 100) {
+          clearInterval(timer.value)
+          progressInternal.value = 100
+        }
+        progressInternal.value = Math.min(progressInternal.value + Math.ceil((Math.random() * 10) * 30), 100)
+      }, 200)
+    })
+
+    onUnmounted(() => {
+      clearInterval(timer.value)
+    })
+
+    return {
+      timer,
+      progressInternal,
+      progression,
+    }
+  },
+})
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles/variables';
+@import '@/styles/functions';
+.fullscreen-loading-container {
+  align-items: center;
+  background: var(--white, color(white));
+  bottom: 0;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  left: 0;
+  margin: var(--KSkeletonFullScreenMargin, 0);
+  position: fixed;
+  right: 0;
+  top: 0;
+  z-index: 10500;
```

Comment:
I believe a lot of these styles are replicated from the other fullscreen loader. Instead of copying over, would it make sense to move them to one of the shared files and then import them in each?